api/v1beta1/common_types.go

```diff
@@ -160,27 +154,19 @@ type IstioControlPlaneReference struct {
 
 // GracefulActionState holds information about GracefulAction State
 type GracefulActionState struct {
-	// ErrorMessage holds the information what happened with CC
-	ErrorMessage string `json:"errorMessage"`
-	// CruiseControlTaskId holds info about the task id ran by CC
-	CruiseControlTaskId string `json:"cruiseControlTaskId,omitempty"`
-	// TaskStarted hold the time when the execution started
-	TaskStarted string `json:"TaskStarted,omitempty"`
-	// CruiseControlState holds the information about CC state
+	// CruiseControlState holds the information about graceful action state
 	CruiseControlState CruiseControlState `json:"cruiseControlState"`
-	// VolumeStates holds the information about the CC disk rebalance states and tasks
+	// CruiseControlOperationReference referencing to the created CruiseControlOperation to execute CC task
+	CruiseControlOperationReference *corev1.LocalObjectReference `json:"cruiseControlOperationReference,omitempty"`
```

Comment:
How are these external `CruiseControlOperation` CRs being managed? Say there are multiple of them created throughout the operation of kafka - how are the old CruiseControlOperation CRs deleted?

Reply:
Not reclaiming them would create a lot of waste in time. +1 for that flag - similar to Job resources they should be by defaukt deleted after a while since they completed successfully

Reply:
Dear @amuraru! Please let me know when you could test it

Reply:
Do you think it would be better to have something like:
ttlAfterCompleted similar to:
https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished/#ttl-after-finished-controller


I am thinking at the usecase where these CRs would be monitored and give the external monitoring a chance to notice a completed operation before beibg deleted

Reply:
It can be solved. Logic also needed in the CruiseControlOperation_controller.

Reply:
I have implemented a new controller for this:
https://github.com/banzaicloud/koperator/pull/865